lib/iris/tests/unit/cube_coord_common/test_CFVariableMixin.py

```diff
@@ -131,6 +131,47 @@ def test_fail_token_default(self):
             self.cf_var.name(default="_nope", token=True)
 
 
+class Test_names(tests.IrisTest):
+    def setUp(self):
+        self.cf_var = CFVariableMixin()
+        self.cf_var.standard_name = None
+        self.cf_var.long_name = None
+        self.cf_var.var_name = None
+        self.cf_var.attributes = dict()
+
+    def test_standard_name(self):
+        standard_name = "air_temperature"
+        self.cf_var.standard_name = standard_name
+        expected = (standard_name, None, None, None)
+        result = self.cf_var.names
+        self.assertEqual(expected, result)
+        self.assertEqual(result.standard_name, standard_name)
+
+    def test_long_name(self):
+        long_name = "air temperature"
+        self.cf_var.long_name = long_name
+        expected = (None, long_name, None, None)
+        result = self.cf_var.names
+        self.assertEqual(expected, result)
+        self.assertEqual(result.long_name, long_name)
+
+    def test_var_name(self):
+        var_name = "atemp"
+        self.cf_var.var_name = var_name
+        expected = (None, None, var_name, None)
+        result = self.cf_var.names
+        self.assertEqual(expected, result)
+        self.assertEqual(result.var_name, var_name)
+
+    def test_STASH(self):
+        stash = "m01s16i203"
+        self.cf_var.attributes = dict(STASH=stash)
+        expected = (None, None, None, stash)
+        result = self.cf_var.names
+        self.assertEqual(expected, result)
+        self.assertEqual(result.STASH, stash)
+
```

Comment:
Should have another testcase here for no names --> (None, None, None, None)
Maybe also a combination case (set them all)

Reply:
@pp-mo Done. A combination is not necessary, that's covered by the permutations of the other tests.

Reply:
:+1: